File: VHBoomMenuButtonDemo/ChangeBoomButtonVC.m
//
//  ChangeBoomButtonVC.m
//  VHBoomMenuButton
//
//  Created by viktorhuang on 2017/4/18.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

#import "ChangeBoomButtonVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface ChangeBoomButtonVC ()<VHBoomDelegate>

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb;

@end

@implementation ChangeBoomButtonVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.bmb.buttonEnum = VHButtonHam;
    self.bmb.piecePlaceEnum = VHPiecePlaceHAM_5;
    self.bmb.buttonPlaceEnum = VHButtonPlaceHAM_5;
    self.bmb.boomDelegate = self;
    self.bmb.autoHide = NO;
    
    VHHamButtonBuilder *builder;
    
    [self.bmb addBuilder:[BuilderManager hamButtonBuilderWithText:@"Change Text" withSubText:@"..."]];
    
    builder = [BuilderManager hamButtonBuilderWithText:@"Change Image" withSubText:@"..."];
    builder.normalImageName = @"eagle";
    [self.bmb addBuilder:builder];
    
    builder = [BuilderManager hamButtonBuilderWithText:@"Change Color" withSubText:@"..."];
    builder.normalColor = [UIColor blueColor];
    [self.bmb addBuilder:builder];
    
    [self.bmb addBuilder:[BuilderManager hamButtonBuilderWithText:@"Change Piece Color" withSubText:@"..."]];
    
    builder = [BuilderManager hamButtonBuilderWithText:@"Change Unable" withSubText:@"..."];
    builder.unableColor = [UIColor grayColor];
    builder.unableImageName = @"butterfly";
    builder.unableText = @"Unable!";
    [self.bmb addBuilder:builder];
}

#pragma mark - VHBoomDelegate

- (void)boomMenuButton:(VHBoomMenuButton *)bmb didClickBoomButtonOfBuilder:(VHBoomButtonBuilder *)builder at:(int)index
{
    VHHamButtonBuilder *hamBuilder = (VHHamButtonBuilder *)[self.bmb builder:index];
    switch (index)
    {
        case 0:
            hamBuilder.normalText = @"Changed!";
            hamBuilder.highlightedText = @"Highlighted, changed!";
            hamBuilder.normalSubText = @"Sub-text, changed!";
            hamBuilder.normalTextColor = [UIColor yellowColor];
            hamBuilder.highlightedColor = [UIColor purpleColor];
            hamBuilder.normalSubTextColor = [UIColor blackColor];
            break;
        case 1:
            hamBuilder.normalImageName = @"bat";
            hamBuilder.highlightedImageName = @"bear";
            break;
        case 2:
            hamBuilder.normalColor = [UIColor redColor];
            break;
        case 3:
            hamBuilder.pieceColor = [UIColor whiteColor];
            break;
        case 4:
            hamBuilder.unable = YES;
            break;
        default:
            break;
    }
}

@end

File: VHBoomMenuButtonSwiftDemo/ChangeBoomButtonVC.swift
//
//  ChangeBoomButtonVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by Nightonke on 2017/5/5.
//  Copyright © 2017 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class ChangeBoomButtonVC: BoomMenuButtonVC, BoomDelegate {

    @IBOutlet weak var bmb: BoomMenuButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        bmb.buttonEnum = .ham
        bmb.piecePlaceEnum = .ham_5
        bmb.buttonPlaceEnum = .ham_5
        bmb.boomDelegate = self
        bmb.autoHide = false
        
        var builder = BuilderManager.hamButtonBuilder(withText: "Change Text", withSubText: "...")
        bmb.addBuilder(builder)
        
        builder = BuilderManager.hamButtonBuilder(withText: "Change Image", withSubText: "...")
        builder.normalImageName = "eagle"
        bmb.addBuilder(builder)
        
        builder = BuilderManager.hamButtonBuilder(withText: "Change Color", withSubText: "...")
        builder.normalColor = UIColor.blue
        bmb.addBuilder(builder)
        
        builder = BuilderManager.hamButtonBuilder(withText: "Change Piece Color", withSubText: "...")
        bmb.addBuilder(builder)
        
        builder = BuilderManager.hamButtonBuilder(withText: "Change Unable", withSubText: "...")
        builder.unableColor = UIColor.gray
        builder.unableImageName = "butterfly"
        builder.unableText = "Unable!"
        bmb.addBuilder(builder)
    }
    
    func boomMenuButton(_ bmb: BoomMenuButton, didClickBoomButtonOfBuilder builder: BoomButtonBuilder, at index: Int) {
        if let hamBuilder = builder as? HamButtonBuilder {
            switch index {
            case 0:
                hamBuilder.normalText = "Changed!"
                hamBuilder.highlightedText = "Highlighted, changed!"
                hamBuilder.normalSubText = "Sub-text, changed!"
                hamBuilder.normalTextColor = UIColor.yellow
                hamBuilder.highlightedColor = UIColor.purple
                hamBuilder.normalSubTextColor = UIColor.black
            case 1:
                hamBuilder.normalImageName = "bat"
                hamBuilder.highlightedImageName = "bear"
            case 2:
                hamBuilder.normalColor = UIColor.red
            case 3:
                hamBuilder.pieceColor = UIColor.white
            case 4:
                hamBuilder.unable = true
            default:
                break
            }
        }
    }

}
